packages/slate-react/src/hooks/use-children.tsx

```diff
@@ -52,7 +52,7 @@ const useChildren = (props: {
     const ds = decorate([n, p])
 
     for (const dec of decorations) {
-      const d = Range.intersection(dec, range)
+      const d = Range.intersection(toAbsoluteRange(dec), range)
```

Comment:
absolute ranges have to be calculated on checking decorations intersection as well